paper/paper.md

```diff
@@ -25,18 +25,91 @@ bibliography: paper.bib
+
+Any one of the four use cases above would serve by itself as a sufficient case for choosing `lintsampler`, but here we give an example of a real-world scenario that combines all of the use cases. It is drawn from our own primary research interests in computational astrophysics.
+
+Much of computational astrophysics consists of large-scale high performance computational simulations of gravitating systems. For example, simulations of planets evolving and interacting within a solar system, simulations of stars interacting within a galaxy, or vast cosmological simulations in which a whole universe is grown __in silico__. 
+
+There exists a myriad of codes used to run these simulations, each using different algorithms to solve the governing equations. One class of simulation code that has gained much attention in recent years is the class of code employing basis function expansions [@agama.paper;@exp.paper]. In these codes, the matter density at any point in space is represented by a sum over basis functions (not unlike a Fourier series), with the coefficients in the sum changing over space and time. As such, the matter comprising the system is represented everywhere as a smooth, continous fluid, but for many applications and/or downstream analyses of the simulated system, one needs to instead represent the system as a set of discrete particles. These particles can be obtained by drawing samples from the continuous density field.
+
+This is a scenario that satisfies all four of the use cases list above. To explain further:
+
+- The PDF we are sampling from (i.e., the basis expansion representation of the matter density field) can be expensive to evaluate if a large number of terms are included in the sum.
+- The PDF can be highly multimodal when the system we are simulating comprises many distinct gravitating substructures, such as stellar clusters.
+- The PDF can have a large dynamic range. Astrophysical structures such as galaxies and dark matter 'haloes' often have power-law density profiles, such as the Navarro-Frenk-White profile [@nfw.paper]. Further complicating the issue is that a typical dark matter halo will host several 'subhaloes', which in turn might host 'subsubhaloes', and so on. In short, a range of spatial scales needs to be resolved.
+- If the particle set being sampled is to be used for further simulation, it can be helpful to draw as 'noiseless' a sample as possible for reasons of numerical stability.
+
+For these reasons, this kind of astrophysical simulation code provides an excellent example of a 'real world' application for `lintsampler`. Here, one would cover the simulation domain with a `DensityTree` instance (or several instances, one for each primary structure), call the `refine` method to better resolve the high-density regions, then feed the tree to a `LintSampler` instance and call `sample` to generate particles. The `qmc` flag can be passed to the sampler in order to employ Quasi-Monte Carlo sampling.
```

Comment:
Hi @matt-graham,

Yes, the basis functions typically have known normalisation constants. However, they do not individually have to be well-behaved probability distributions[^1].

[^1]:And often are not, instead chosen such that the lowest-order function resembles the density and higher-order functions are perturbations with positive and negative regions.

In a bit more detail: in the most-convenient usage, the basis functions are sets of ortho-_normal_ polynomials[^2], satisfying $\delta_{mn} = \int \phi_m(x,\theta_n)\phi_n(x,\theta_n)dx$. 

[^2]: It is possible to orthonormalise any set of functions, but it is just easier if we know the normalisation (=inner product of each function with itself) a priori.

Then, given $P$ uniform-weight sample points $[x_1,x_2,...x_P]$, the weights $w_n$ are computed as 

$$w_n = \frac{1}{P} \sum_{p=1}^P \phi_n(x_p,\theta_n),$$

which could be considered the approximation to the integral

$$Z_n(\theta_n) = \int \phi_n(x, \theta_n) \\,\mathrm{d}x$$

with sample points representing the density as 

$$p(x)=\sum_{p=1}^{P}\delta(x-x_p).$$

For the case of `lintsampler`, one could 'close the loop' and measure discreteness noise by drawing samples from a target 

$$p(x)=\sum_{n=1}^{N} w_n \phi_n(x, \theta_n)$$ 

where the weights are known, and then checking the recovery of $w_n$ as above.

Reply:
Hi @michael-petersen, thanks for the additional details and apologies for my delay in responding. I think the key point I was missing initially was the point that the basis functions need not individually correspond to valid probability density functions as they can be negative valued in some regions. My initial thought on reading the description in the paper had been that a natural way of sampling from the target distribution would be to use an approach akin to sampling from a Gaussian mixture of first sampling indices of mixture components and then sampling from the component distributions (as this would avoid the issues mentioned of having to evaluate all terms in sum to evaluate the PDF and typically would also help alleviate issues with multimodality) however I can now see this would not be a valid approach in this scenario.